### ISSUE TITLE
Multiplayer shows In Room is True even after being Kicked or Disconnecting from singaling server.

### ISSUE THREAD

**Issue by Alaadel** [Scirra/Construct-3-bugs#4593]
## Problem description

The multiplayer plugin still indicates I am in a room even after being kicked, or disconnected from the signaling server. This can cause the players to not be able to leave, not be able to host their own game, and not be even able to rejoin the original game.

## Attach a .c3p
[Test Multiplayer S.zip](https://github.com/Scirra/Construct-3-bugs/files/6017235/Test.Multiplayer.S.zip)

## Steps to reproduce

1. Run 2 debug previews.
**In the first preview (host):**
2. Connect.
3. Login (after it connects).
4. Join (after it logs in).
**In the second preview (peer):**
5. Connect
6. Log in (after it connects).
7. Join (after it logs in).
8. Copy the Peer's ID from its text input (the 5th text input).
**In the first preview (host):**
9. Paste it in the kick text input on the Host side.
10. Click "Kick".
**In the second preview (peer):**
11. Additionally, click "DC Server" on the Peer side. It will still show the Peer in Room.
12. Click Leave.
13. Change the room name to "123456" (to force the peer to create a new room and become a host).
14. Click Connect.
15. Click Login (after it connects).
16. Click Join (after it logs in).

## Observed result

After step 10:
- The multiplayer shows I am in room.
- "On Kicked" event does not fire on the Peer.
- The Chrome console shows "RTCErrorEvent" on both Peer and Host. (I have attached the peer and host logs from the console: [Logs.zip](https://github.com/Scirra/Construct-3-bugs/files/6017254/Logs.zip) )

After step 12:
- The peer will show "leaving..." but it will never leave, and will not show any error to indicate that leaving in this state is not possible.

After step 16:
- The peer will show "joining..." but will never join, even though it is connected and logged in to the server.

## Expected result

- After being kicked, disconnected from the server, or logged out: The multiplayer shows I am Not in room.
- "On Kicked" event should fire on the Peer after being kicked by the host (additionally, the peer would receive the kick reason inside Multiplayer.Message). And the peer should be removed automatically from the room.

## More details

The solution is simply for the peer to Leave the room before Leaving the server. The peer needs to leave (by himself) even after being Kicked by the host, otherwise, it seems it will still be considered in a room.

Alternatively, you can repeat the above steps, but after step 10, click "Leave" on the peer side, then complete the rest of the steps and you won't face the problem; except (after the peer becomes a host, after step 16) the console logs will show errors on both sides again.

Also, the "Kicked!" event fires on the Peer if the Host clicked "Leave".

**Affected browsers/platforms:** Chrome (88.0.4324.182 (Official Build) (64-bit))

When testing on Firefox (85.0.2 (64-bit)):
- The In Room shows correct results, except after logging in, it shows "In Room" even before clicking "Join". i.e. it shows "In Room" after step 3 instead of 4. But on Chrome, it correctly shows "In Room" after step 4.
- Correctly shows "Kicked!" after being kicked by the host.
- After step 12, it still shows "Leaving...", and never shows "Left." But now the Multiplayer shows I am not even in a room and not in the server, so this result is understandable, specially as it does not block the rest of the steps, and I can host or reconnect again.
- No console errors.

So, to sum up the findings: Signaling works more accurately and smoothly on Firefox, (aside from showing "In Room" right after "Login" even before clicking "Join").

**First affected release:** 
Only tested on r234.4 stable.

## System details

<details><summary>View details</summary>

PASTE HERE

</details>

**Comment by AshleyScirra** [comment]
The room check is incorrect because of a mistake in your events: event 39 makes the wrong check (is logged in rather than is in room).

The real problem is "On kicked" not firing, which causes the multiplayer state to get stuck because it still thinks it's in the room and will stay like that for a period of time until the connection times out. This looks like a Chrome bug, since as noted it works in Firefox: it seems since we send a kick message and then immediately close the connection, Chrome doesn't send anything out to the network; if we add a small delay before closing, it works. I couldn't reproduce it in a minimal example though so it's difficult to file anything with Google about it. The workaround I added seems to fix it though, so it should be working in the next beta.

I don't think the console errors matter much - if you terminate a connection with messages in-flight, it will tend to log a couple of errors about the transmission failing due to the connection being terminated, but that's not otherwise a problem as long as that last kick message is dispatched.

**Comment by Alaadel** [comment]
Thank you.
If I skip the kick step (9 and 10) and click "DC Server" on peer side, it leaves server and logs out, but In Room still shows In room (sorry about the bug, I forgot to update the sample here after I fixed it earlier). Is this by design? i.e if peer is not connected to the server, then the data in "In Room" is irrelevant?

I wanted to ask first before creating a new issue...
